### PR TITLE
Callout ECR not being supported

### PR DESCRIPTION
**Pull request by stuart23**
ECR is not a supported container registry for Astro due to its frequent key rotations that we are unable to handle.

### CODE REVIEW

**Comment by stuart23** [comment]
@paolaperaza we discussed this 2 weeks ago (I think). It has just come up with a customer as well 
https://astronomer.zendesk.com/agent/tickets/11433

File: astro/kubernetespodoperator.md [hunk 1]
```diff
@@ -102,6 +102,12 @@ Applying the code above ensures that when this DAG runs, it will launch a Kubern
 
 By default, the KubernetesPodOperator expects to pull a Docker image that's hosted publicly on Docker Hub. If you want to execute a Docker image that's hosted in a private registry, complete the setup below.
 
+:::caution
+
+AWS Elastic Container Registry (ECR) is currently not supported on Astro. ECR credentials expire after 12 hours and there is no way for Astro to create a new set of credentials.
```
Comment (kwcanuck) [inline astro/kubernetespodoperator.md:107]:
```suggestion
Amazon Elastic Container Registry (ECR) is not supported. ECR credentials expire after 12 hours and they can't be recreated.
```

**Review by kwcanuck — APPROVED** [review]

File: astro/kubernetespodoperator.md [hunk 2]
```diff
@@ -102,6 +102,12 @@ Applying the code above ensures that when this DAG runs, it will launch a Kubern
 
 By default, the KubernetesPodOperator expects to pull a Docker image that's hosted publicly on Docker Hub. If you want to execute a Docker image that's hosted in a private registry, complete the setup below.
 
+:::caution
+
+Amazon Elastic Container Registry (ECR) is not supported. ECR credentials expire after 12 hours and they can't be recreated.
```
Comment (paolaperaza) [inline astro/kubernetespodoperator.md:107]:
```suggestion
Running Docker images that are hosted on [Amazon Elastic Container Registry](https://aws.amazon.com/ecr/) (ECR) is not supported on Astro. Authentication tokens generated by ECR expire after 12 hours and cannot be recreated, which means that KubernetesPodOperator tasks would not reliably have the necessary credentials to pull and run any Docker image hosted on ECR.

If you're using ECR and are interested in exploring other solutions, contact [Astronomer support](https://support.astronomer.io).
```

**Review by paolaperaza — APPROVED** [review]
This is my official recommendation! @stuart23 @kwcanuck LMK what you guys think

File: astro/kubernetespodoperator.md [hunk 3]
```diff
@@ -102,6 +102,14 @@ Applying the code above ensures that when this DAG runs, it will launch a Kubern
 
 By default, the KubernetesPodOperator expects to pull a Docker image that's hosted publicly on Docker Hub. If you want to execute a Docker image that's hosted in a private registry, complete the setup below.
 
+:::caution
+
+Running Docker images that are hosted on [Amazon Elastic Container Registry](https://aws.amazon.com/ecr/) (ECR) is not supported on Astro. Authentication tokens generated by ECR expire after 12 hours and cannot be recreated, which means that KubernetesPodOperator tasks would not reliably have the necessary credentials to pull and run any Docker image hosted on ECR.
+
+If you're using ECR and are interested in exploring other solutions, contact [Astronomer support](https://support.astronomer.io).
```
Comment (paolaperaza) [inline astro/kubernetespodoperator.md:109]:
```suggestion
Running Docker images that are hosted on [Amazon Elastic Container Registry](https://aws.amazon.com/ecr/) (ECR) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks cannot have the credentials necessary to pull and run Docker images hosted on ECR.

If you're using ECR and are interested in exploring other solutions, contact [Astronomer support](https://support.astronomer.io).
```

File: astro/kubernetespodoperator.md [hunk 4]
```diff
@@ -102,6 +102,14 @@ Applying the code above ensures that when this DAG runs, it will launch a Kubern
 
 By default, the KubernetesPodOperator expects to pull a Docker image that's hosted publicly on Docker Hub. If you want to execute a Docker image that's hosted in a private registry, complete the setup below.
 
+:::caution
+
+Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks cannot have the credentials necessary to pull and run Docker images hosted on ECR.
+
+If you're using ECR and are interested in exploring other solutions, contact [Astronomer support](https://support.astronomer.io).
```
Comment (paolaperaza) [inline astro/kubernetespodoperator.md:109]:
```suggestion
Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks don't have the credentials necessary to pull and run Docker images hosted on ECR.

If you're using ECR and are interested in exploring other solutions, contact [Astronomer support](https://support.astronomer.io).
```

Comment (stuart23) [reply astro/kubernetespodoperator.md]: Technically the first option is more correct

File: astro/kubernetespodoperator.md [hunk 5]
```diff
@@ -102,6 +102,14 @@ Applying the code above ensures that when this DAG runs, it will launch a Kubern
 
 By default, the KubernetesPodOperator expects to pull a Docker image that's hosted publicly on Docker Hub. If you want to execute a Docker image that's hosted in a private registry, complete the setup below.
 
+:::caution
+
+Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks don't have the credentials necessary to pull and run Docker images hosted on ECR.
```
Comment (stuart23) [inline astro/kubernetespodoperator.md:107]:
```suggestion
Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, Kubelet will not have the credentials necessary to pull and run Docker images hosted on ECR in KubernetesPodOperator tasks.
```

Comment (stuart23) [reply astro/kubernetespodoperator.md:107]: Technically this is more correct but might confuse people

Comment (paolaperaza) [inline astro/kubernetespodoperator.md:107]:
```suggestion
Running Docker images that are hosted on [Amazon Elastic Container Registry (ECR)](https://aws.amazon.com/ecr/) isn't supported on Astro. ECR authentication tokens expire after 12 hours and can't be recreated. As a result, KubernetesPodOperator tasks don't have access to the credentials necessary to pull and run Docker images hosted on ECR.
```

Yea, I think that's too low-level -- @stuart23 this better?

Comment (stuart23) [reply astro/kubernetespodoperator.md:107]: The intent is ok. I don't think people are going to pull us up on it, so I don't mind if you want to keep it more readable.

Comment (kwcanuck) [reply astro/kubernetespodoperator.md]: Maybe:

"...can't access the credentials necessary..."

**Pull request merged**